core/src/main/java/com/datastax/oss/driver/api/core/config/DefaultDriverOption.java

```diff
@@ -808,6 +808,20 @@
    * <p>Value-type: int
    */
   SESSION_LEAK_THRESHOLD("advanced.session-leak.threshold"),
+
+  /**
+   * The fully-qualified classname of the desired MetricsFactory implementation.
+   *
+   * <p>Value-type: {@link String}
+   */
+  METRICS_FACTORY_CLASS("advanced.metrics.factory.class"),
+
+  /**
+   * The fully-qualified classname of the desired metrics registry implementation.
+   *
+   * <p>Value-type: {@link String}
+   */
+  METRICS_FACTORY_REGISTRY_CLASS("advanced.metrics.factory.registry.class"),
```

Comment:
do we really need those two parameters? I think that `METRICS_FACTORY_CLASS` should be enough.
You wrote:
```
https://github.com/datastax/java-driver/pull/1464/files#diff-8b8ff589cb369ea902b39dcf4601a449R1242-R1245
```
could you elaborate more on that?

Reply:
My explanation is long, the TL;DR is this:
I wanted to try to avoid having a hard-coded metrics registry implementation for MicroProfile (i.e. smallrye). I'm not sure it's worth the effort.

Longer explanation:
I added this to allow for configuring a class that could be instantiated as a metric registry, mostly for the case of MicroProfile. Each metrics framework (DropWizard, Micrometer and MicroProfile) has a registry for holding the collected metrics, and each requires access to that registry in order to create the desired metrics.

In the case of DropWizzard and Micrometer, a default or global registry is provided with the framework, so the MetricsFactory implementation for those frameworks does not need a configured registry class.

DropWizard: https://github.com/datastax/java-driver/pull/1464/files#diff-a204e0c3b6601fba49c1b8f3414563cbR69

Micrometer: https://github.com/datastax/java-driver/pull/1464/files#diff-7823a53975fd907c9990f1e6a327fd56R71

But for MicroProfile, there is no included concrete implementation of a MeterRegistry. We use Smallye for the implementation, but there are other libraries that could be used. Currently, I have the registry hardcoded here:

https://github.com/datastax/java-driver/pull/1464/files#diff-c3b4e1da8e315c17af71d25d32983dafR71

However, if the registry class is configurable, I can remove the Smallrye dependency and allow users to configure whatever registry they want to use. But as I'm typing this explanation, I'm realizing that if we want this MicroProfile module to be drop-in, it will need an implementation of the registry, so smallrye will have to be included.

So the only reason to keep the registry class configurable is to allow for different implementations of a given metrics framework's registry, which may not really be applicable outside of MicroProfile anyway.

Reply:
I agree that two options sounds a bit complicated. Maybe there should be one factory per available MicroProfile implementation, e.g. `MicroprofileSmallRyeMetricsFactory implements MetricsFactory`. Are there any other implementation than SmallRye out there, and is it likely that we'd need to support them?

Reply:
> Are there any other implementation than SmallRye out there, and is it likely that we'd need to support them?

I think this is the list: https://wiki.eclipse.org/MicroProfile/Implementation#MP_Metrics_implementations
So just 2 others that implement the latest version of MicroProfile metrics.

Reply:
So it seems the other two MicroProfile implementations are other apps, not necessarily libraries intended to be used as a metrics implementation. So, it looks like smallrye is the only one we will need to handle for now.